src/components/Tamagotchi.tsx

```diff
@@ -1,28 +1,70 @@
 "use client";
 import { INTERACTION } from "@/app/utils/interaction";
 import React, { useState, useEffect } from "react";
-import { idle, drinkingCoffee } from "./tamagotchiFrames";
+import { idle, drinkingCoffee, death } from "./tamagotchiFrames";
 
 import "../index.css";
 
 const DEFAULT_STATUS = ":)";
 const Tamagotchi: React.FC = () => {
   const [frameIndex, setFrameIndex] = useState<number>(0);
-  const [animation, setAnimation] = useState(idle);
+  const [tamagotchiState, setTamagotchiState] = useState<any>({});
+  const [animation, setAnimation] = useState<string[]>(idle);
   const [tamaStatus, setTamaStatus] = useState(DEFAULT_STATUS);
+  const [checkingStatus, setCheckingStatus] = useState(false);
+  const [isInteracting, setIsInteracting] = useState(false);
 
   //TODO - call init endpoint to determine if tamagotchi is initialized. if not generate one.
 
   useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch("/api/getState", { method: "POST" });
+        if (response.ok) {
+          const jsonData = await response.json();
+          setTamagotchiState(jsonData);
+          handleDeath(jsonData, pollInterval);
+        }
+      } catch (error) {
```

Comment:
**suggestion:** Consider showing an error state to the user when the fetch fails

Currently errors are only logged to the console. Add error state handling to show appropriate feedback to users when data fetching fails.

```
      } catch (error) {
        setError('Failed to fetch Tamagotchi data');
        setIsLoading(false);
```